src/server.rs

```diff
@@ -130,7 +130,7 @@ impl Server {
             Some(ClientMessage::Accept(id)) => {
                 info!(%id, "forwarding connection");
                 match self.conns.remove(&id) {
-                    Some((_, stream2)) => proxy(stream, stream2).await?,
+                    Some((_, stream2)) => proxy(stream.get_mut(), stream2).await?,
```

Comment:
I wonder if it makes more sense to take `stream.into_inner()` here instead, since we're not using the stream afterward.